script.js

```diff
@@ -1,12 +1,547 @@
+
+  const selectOption = document.createElement("option");
+  selectOption.value = "select-episode";
+  selectOption.textContent = "Select Episode";
+  episodeSelect.appendChild(selectOption);
+
+  episodes.forEach((episode) => {
+    const option = document.createElement("option");
+    option.value = episode.id;
+    option.textContent = `${episode.name} - S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
+    episodeSelect.appendChild(option);
+  });
+
+  episodeSelectContainer.appendChild(episodeSelect);
+  searchSelectDiv.appendChild(episodeSelectContainer);
+
+  hideShowSearchField(); 
+  // Hiding the search field for shows in the episodes display
+}
+
+// Level 200 - Adding search field for episodes
+function addSearchField() {
+  const searchSelectDiv = document.querySelector(".search-select");
+
+  searchContainer = document.createElement("div");
+  searchContainer.classList.add("search-container");
+
+  const searchForm = document.createElement("form");
+  searchForm.classList.add("search-form");
+
+  const searchInput = document.createElement("input");
+  searchInput.classList.add("search-input");
+  searchInput.type = "search";
+  searchInput.placeholder = "Search episodes";
+  searchInput.id = "search-input";
+
+  const searchLabel = document.createElement("label");
+  searchLabel.classList.add("search-label");
+  searchLabel.setAttribute("for", "search-input");
+
+  searchForm.appendChild(searchInput);
+  searchForm.appendChild(searchLabel);
+  searchContainer.appendChild(searchForm);
+  searchSelectDiv.appendChild(searchContainer);
+
+  // adding event listener to the search input
+  searchInput.addEventListener("input", () => {
+    const query = searchInput.value.toLowerCase();
+    const filteredEpisodes = allEpisodes.filter((episode) => {
+      const title = `${episode.name} - S${episode.season
+        .toString()
+        .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
+      const titleMatch = title.toLowerCase().includes(query);
+      const summaryMatch = episode.summary.toLowerCase().includes(query);
+      return titleMatch || summaryMatch;
+    });
+
+    // updating the episode list with the filtered episodes
+    makePageForEpisodes(filteredEpisodes);
+
+    // updating the search label with the number of filtered episodes
+    searchLabel.textContent = `Displaying ${filteredEpisodes.length} / ${allEpisodes.length} episodes`;
+  });
+
+  // setting the initial search label text
+  searchLabel.textContent = `Displaying ${allEpisodes.length} / ${allEpisodes.length} episodes`;
+}
+
+
+
+//Creating the header of page
+function createHeader() {
+  const body = document.body;
+
+  const header = document.createElement("header");
+  header.classList.add("header");
+  body.insertBefore(header, body.firstChild);
+
+  const titleHeader = document.createElement("h1");
+  titleHeader.classList.add("title");
+  titleHeader.textContent = "Cineflix";
```

Comment:
it would be nice if you could click this header to go back to the initial choose shows page

Reply:
Yes, I was thinking about it to be honest as back to show button would take more time but I don’t know, how would users understand that is clickable :/